### PR TITLE
Enable explicit API mode

### LINKED ISSUE #228: Enable "library mode"
https://kotlinlang.org/docs/whatsnew14.html#explicit-api-mode-for-library-authors

### PR DESCRIPTION
**Pull request by rock3r**
This PR enables the explicit API mode in the Kotlin compiler. This means that all public API must be explicitly annotated as such, and provide explicit return types, or the compilation will break. This ensures better API stability and avoids unexpected return types (it already caught something, which is fixed in this PR).

Since this PR is _immense_, know that it is:
 1. Enabling the explicit API mode
 2. Adding public modifiers and explicit return types until the project runs again
 3. Reformatting _everything_
 4. Cleaning up some minor style problems in code that was changed anyway
 5. Providing better error messages for missing composition locals, that point to the lack of theme (that code was impacted anyway from the rest of the changes)

Closes #228

### CODE REVIEW

**Comment by rock3r** [comment]
Switching reviewer to @devkanro since @lamba92 is on vacation

**Review by devkanro — CHANGES_REQUESTED** [review]
The indents need be fixed.

**Comment by rock3r** [comment]
All fixed :)

File: buildSrc/src/main/kotlin/IdeaConfiguration.kt [hunk 1]
```diff
@@ -11,17 +11,21 @@ enum class SupportedIJVersion {
 private var warned = AtomicBoolean(false)
 
 fun Project.supportedIJVersion(): SupportedIJVersion {
-    val prop = kotlin.runCatching {
-        rootProject.findProperty("supported.ij.version")?.toString() ?:
-            localProperty("supported.ij.version")
-    }.getOrNull()
+    val prop =
+        kotlin
+            .runCatching {
```
Comment (devkanro) [inline buildSrc/src/main/kotlin/IdeaConfiguration.kt:16]: This line breaks seem unnecessary.

File: buildSrc/src/main/kotlin/jewel-ij-publish.gradle.kts [hunk 2]
```diff
@@ -29,24 +31,23 @@ publishing {
             from(components["kotlin"])
             artifact(javadocJar)
             artifact(sourcesJar)
-            val ijVersionRaw = when (supportedIJVersion()) {
-                IJ_232 -> "232"
-                IJ_233 -> "233"
-            }
+            val ijVersionRaw =
+                when (supportedIJVersion()) {
```
Comment (devkanro) [inline buildSrc/src/main/kotlin/jewel-ij-publish.gradle.kts:35]: Also this

File: buildSrc/src/main/kotlin/jewel-publish.gradle.kts [hunk 3]
```diff
@@ -6,13 +6,15 @@ plugins {
     id("org.jetbrains.dokka")
 }
 
-val sourcesJar by tasks.registering(Jar::class) {
+val sourcesJar by
+tasks.registering(Jar::class) {
```
Comment (devkanro) [inline buildSrc/src/main/kotlin/jewel-publish.gradle.kts:10]: Also this

File: buildSrc/src/main/kotlin/jewel.gradle.kts [hunk 4]
```diff
@@ -6,14 +6,15 @@ plugins {
...
-version = when {
-    gitHubRef?.startsWith("refs/tags/") == true -> {
-        gitHubRef.substringAfter("refs/tags/")
-            .removePrefix("v")
-    }
 
-    else -> "1.0.0-SNAPSHOT"
-}
+version =
+    when {
```
Comment (devkanro) [inline buildSrc/src/main/kotlin/jewel.gradle.kts:11]: This

File: buildSrc/src/main/kotlin/org/jetbrains/jewel/buildlogic/demodata/AndroidStudioReleasesReader.kt [hunk 5]
```diff
@@ -83,19 +94,23 @@ internal object AndroidStudioReleasesReader {
...
+                .trim()
+                .replace(" ", "")
 
         if (releaseAnimal.isEmpty() || releaseAnimal.any { it.isDigit() }) return null
 
         // We only have stable and canary splash screens. Betas use the stable ones.
-        val channel = release.channel.lowercase()
-            .let {
+        val channel =
+            release.channel.lowercase().let {
```
Comment (devkanro) [inline buildSrc/src/main/kotlin/org/jetbrains/jewel/buildlogic/demodata/AndroidStudioReleasesReader.kt:113]: This

File: buildSrc/src/main/kotlin/org/jetbrains/jewel/buildlogic/theme/IntUiThemeDescriptorReader.kt [hunk 6]
```diff
@@ -23,68 +24,77 @@ internal object IntUiThemeDescriptorReader {
...
 
     private fun TypeSpec.Builder.readColors(colors: Map<String, String>) {
-        val colorGroups = colors.entries.groupBy {
-            it.key.replace("""\d+""".toRegex(), "")
-        }.filterKeys {
-            colorGroups.contains(it)
-        }.map { (groupName, colors) ->
-            // We assume color lists are in the same order as in colorGroups
+        val colorGroups =
+            colors.entries
```
Comment (devkanro) [inline buildSrc/src/main/kotlin/org/jetbrains/jewel/buildlogic/theme/IntUiThemeDescriptorReader.kt:70]: This

File: buildSrc/src/main/kotlin/org/jetbrains/jewel/buildlogic/theme/IntelliJThemeGeneratorPlugin.kt [hunk 7]
```diff
@@ -57,14 +59,18 @@ open class IntelliJThemeGeneratorTask : DefaultTask() {
...
         logger.lifecycle("Fetching theme descriptor from $url...")
-        val themeDescriptor = URL(url).openStream()
-            .use { json.decodeFromStream<IntellijThemeDescriptor>(it) }
+        val themeDescriptor =
+            URL(url).openStream().use { json.decodeFromStream<IntellijThemeDescriptor>(it) }
 
         val className = ClassName.bestGuess(themeClassName.get())
-        val file = IntUiThemeDescriptorReader.readThemeFrom(themeDescriptor, className, ideaVersion.get(), url)
+        val file =
+            IntUiThemeDescriptorReader.readThemeFrom(themeDescriptor, className, ideaVersion.get(), url)
```
Comment (devkanro) [inline buildSrc/src/main/kotlin/org/jetbrains/jewel/buildlogic/theme/IntelliJThemeGeneratorPlugin.kt:67]: This

File: decorated-window/src/main/kotlin/org/jetbrains/jewel/window/DecoratedWindow.kt [hunk 8]
```diff
@@ -197,77 +211,79 @@ private object DecoratedWindowMeasurePolicy : MeasurePolicy {
 
         for (it in measurables) {
             if (it.layoutId.toString().startsWith(TITLE_BAR_COMPONENT_LAYOUT_ID_PREFIX)) continue
-            val placeable = it.measure(contentConstraints.offset(vertical = -titleBarHeight - titleBarBorderHeight))
+            val placeable =
+                it.measure(contentConstraints.offset(vertical = -titleBarHeight - titleBarBorderHeight))
```
Comment (devkanro) [inline decorated-window/src/main/kotlin/org/jetbrains/jewel/window/DecoratedWindow.kt:215]: This

File: foundation/src/main/kotlin/org/jetbrains/jewel/foundation/lazy/SelectableLazyColumn.kt [hunk 9]
```diff
@@ -99,76 +87,83 @@ fun SelectableLazyColumn(
...
+                            }
+                        } else {
                             entry.itemContent.invoke(itemScope, index)
                         }
-                    } else {
-                        entry.itemContent.invoke(itemScope, index)
                     }
-                }
+                is Entry.StickyHeader ->
+                    stickyHeader(entry.key, entry.contentType) {
```
Comment (devkanro) [inline foundation/src/main/kotlin/org/jetbrains/jewel/foundation/lazy/SelectableLazyColumn.kt:145]: Maybe this

**Review by devkanro — COMMENTED** [review]
Some line breaks seems unnecessary, can you recheck those style? But I don't block this.

**Comment by rock3r** [comment]
Hopefully it's all fixed for real now :P

**Review by devkanro — APPROVED** [review]
LGTM

**Pull request merged**